maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/PluginOverviewRenderer.java

```diff
@@ -0,0 +1,392 @@
+
+    private final List<RequirementsHistory> requirementsHistories;
+
+    private final PluginDescriptor pluginDescriptor;
+
+    private final boolean hasExtensionsToLoad;
+
+    /**
+     * @param sink                  not null
+     * @param i18n                  not null
+     * @param locale                not null
+     * @param project               not null
+     * @param requirementsHistories not null
+     * @param pluginDescriptor      not null
+     */
+    PluginOverviewRenderer(
+            Sink sink,
+            I18N i18n,
+            Locale locale,
+            MavenProject project,
+            List<RequirementsHistory> requirementsHistories,
+            PluginDescriptor pluginDescriptor,
+            boolean hasExtensionsToLoad) {
+        super(sink, locale, i18n, project);
+
+        this.requirementsHistories = requirementsHistories;
+
+        this.pluginDescriptor = pluginDescriptor;
+
+        this.hasExtensionsToLoad = hasExtensionsToLoad;
+    }
+
+    @Override
+    public String getTitle() {
+        return getI18nString("title");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void renderBody() {
+        startSection(getTitle());
+
+        if (!(pluginDescriptor.getMojos() != null && pluginDescriptor.getMojos().size() > 0)) {
+            paragraph(getI18nString("goals.nogoal"));
+            endSection();
+            return;
+        }
+
+        paragraph(getI18nString("goals.intro"));
+
+        boolean hasMavenReport = false;
+        for (MojoDescriptor mojo : pluginDescriptor.getMojos()) {
+            if (PluginUtils.isMavenReport(mojo.getImplementation(), project)) {
+                hasMavenReport = true;
+            }
+        }
+
+        startTable();
+
+        String goalColumnName = getI18nString("goals.column.goal");
+        String isMavenReport = getI18nString("goals.column.isMavenReport");
+        String descriptionColumnName = getI18nString("goals.column.description");
+        if (hasMavenReport) {
+            tableHeader(new String[] {goalColumnName, isMavenReport, descriptionColumnName});
+        } else {
+            tableHeader(new String[] {goalColumnName, descriptionColumnName});
+        }
+
+        List<MojoDescriptor> mojos = new ArrayList<>();
+        mojos.addAll(pluginDescriptor.getMojos());
+        PluginUtils.sortMojos(mojos);
+        for (MojoDescriptor mojo : mojos) {
+            String goalName = mojo.getFullGoalName();
+
+            /*
+             * Added ./ to define a relative path
+             * @see AbstractMavenReportRenderer#getValidHref(java.lang.String)
+             */
+            String goalDocumentationLink = "./" + mojo.getGoal() + "-mojo.html";
+
+            String description;
+            if (StringUtils.isNotEmpty(mojo.getDeprecated())) {
+                description = "<strong>" + getI18nString("goal.deprecated") + "</strong> " + mojo.getDeprecated();
```

Comment:
This is a bit ugly and should rather not use then `tableCell()`, but use `Sink#tableCell()`/`Sink#tableCell()_`.

Reply:
Everything in PluginOverviewRenderer has just been moved from the inner class in https://github.com/apache/maven-plugin-tools/blob/master/maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/PluginReport.java#L262. The only refactoring was due to the newly introduced common super class (`AbstractPluginReportRenderer`). Let us track other improvements to the `PluginOverviewRenderer` separately.

Reply:
Agreed. Please make a note.